### DISCUSSION TITLE
tableau server spark SQL ODBC driver连接Kuuubi 执行SQL异常

### DISCUSSION THREAD

**Discussion by CCweixiao** [apache/kyuubi#1244] [Q&A]
kyuubi的版本是1.3.0-incubating，引擎共享级别是CONNECTION，

Tableau server上发布的Kyuubi数据源刷新任务，运行刷新任务，提交SQL到Kyuubi server上后，Spark引擎可以正常创建，但会因异常断开导致SQL执行失败。

Tableau server在Windows Server上部署，以spark sql  ODBC driver的方式连接kyuubi，失败的数据源刷新任务报错信息中并无有用信息。

![image-20211015204050669](https://leo-jie-pic.oss-cn-beijing.aliyuncs.com/leo_blog/2021-10-15-124051.png)

Kyuubi server上的具体异常日志信息是：

![image-20211015210043326](https://leo-jie-pic.oss-cn-beijing.aliyuncs.com/leo_blog/2021-10-15-130043.png)

对应源码：

```scala
// org/apache/kyuubi/cli/HandleIdentifier.scala
  
   def apply(tHandleId: THandleIdentifier): HandleIdentifier = {
    info(s"leo >>>>>>>>>> ${tHandleId.toString}")
    val pbb = ByteBuffer.wrap(tHandleId.getGuid)
     // guid为空, secret也为空时 数据转换发生异常
    val publicId = new UUID(pbb.getLong, pbb.getLong)

    val sbb = ByteBuffer.wrap(tHandleId.getSecret)
    val secretId = new UUID(sbb.getLong, sbb.getLong)

    new HandleIdentifier(publicId, secretId)
  }
```

SQL被正常执行时的日志输出：

![image-20211015211231836](https://leo-jie-pic.oss-cn-beijing.aliyuncs.com/leo_blog/2021-10-15-131232.png)

**Comment by yaooqinn** [comment]
did you observe such behavior for some other clients？What the the underlying client for tableau to use

**Reply by CCweixiao** [reply]
> 您是否观察到其他一些客户端的这种行为？ tableau 使用的底层客户端是什么

java jdbc没这个问题，tableau使用的是window 下面的ODBC，其源码没找到，只有一个exe driver驱动程序

**Reply by yaooqinn** [reply]
any information about versions of the stuffs you use so far
